jcl/src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -3866,32 +3961,46 @@ private static MethodHandle recreateIteratedBodyHandle(MethodHandle iteratorHand
 		MethodHandle loopBody = bodyHandle;
 		MethodType bodyType = loopBody.type;
 		Class<?> bodyReturnType = bodyType.returnType;
+		Class<?>[] bodyParamTypes = bodyType.parameterArray();
+		int bodyParamLength = bodyParamTypes.length;
+		int bodyIterationVarLength = 1;
 		
 		if (void.class != bodyReturnType) {
-			Class<?>[] bodyParamTypes = bodyType.parameterArray();
 			Class<?> paramType1 = bodyParamTypes[0];
 			Class<?> paramType2 = bodyParamTypes[1];
 			bodyParamTypes[0] = paramType2;
 			bodyParamTypes[1] = paramType1;
 			MethodType newBodyType = MethodType.methodType(bodyReturnType, bodyParamTypes);
-			loopBody = MethodHandles.permuteArguments(loopBody, newBodyType, 1, 0);
 			
-			/* Insert the external parameter types if the loop body only has V and T */
-			if (2 == bodyParamTypes.length) {
-				/* Insert the class Iterable to the external parameter part of the loop body if iterator is null */
-				if (null == iteratorHandle) {
-					loopBody = MethodHandles.dropArguments(loopBody, 2, Iterable.class);
-				} else {
-					/* Append all parameter types of iterator to the end of the parameter types of loop body */
-					Class<?>[] bodyExternalParameterTypes = iteratorHandle.type.parameterArray();
-					loopBody = MethodHandles.dropArguments(loopBody, 2, bodyExternalParameterTypes);
-				}
+			int[] reorder = new int[bodyParamLength];
```

Comment:
Better to use a more self-explanatory name like "paramIndexRecoder" or something similar.